### PR TITLE
testing

### PR DESCRIPTION
**Pull request by srslie**
**Changes**
- added tests for all components, user stories
- added/changed classsNames and organization as needed
- fixed routing bug on 404, tested
- added api error throw and test

**Issues**
- will need to update baseURL when deployed

- [x] DRY and SRP
- [x] tested locally
- [x] accessibility audited (wave)

**Notes**
- To test, pull down branch with git clone on this branch
-  run ```npm install``` to get dev dependencies 
- and ```npm start``` to use local server
- Then run cypress with ```npx cypress open```

### CODE REVIEW

**Comment by srslie** [comment]
@deadbelly

File: cypress/support/commands.js [hunk 1]
```diff
@@ -23,3 +23,15 @@
...
+//   const opts = Object.assign({}, options = {}, {
+//     onBeforeLoad: (window, ...args) => {
+//       window.fetch = null;
+//       if (options.onBeforeLoad) {
+//         return options.onBeforeLoad(window, ...args);
+//       }
+//     },
+//   });
+//   return originalFn(url, opts);
+// });
```
Comment (deadbelly) [inline cypress/support/commands.js:37]: Maybe take this out if you aren't using it?

File: cypress/integration/examples/app.spec.js [hunk 2]
```diff
@@ -1,23 +1,176 @@
...
\ No newline at end of file
+})
+
+describe('Failure', () => {
+  it.only('Should show an error page if fetch fails', () => {
+    cy.intercept(
+      'GET', 
+      apiUrl,
+      { statusCode: 400}
+    )
```
Comment (deadbelly) [inline cypress/integration/examples/app.spec.js:168]: Could be worth testing at least one other kind of error

**Review by deadbelly — APPROVED** [review]
This is looking pretty good. I made a couple suggestions for changes but I think it's solid even without those. Good job and reworking how your errors rendered.

File: cypress/integration/examples/app.spec.js (hunk 2 above)
Comment (srslie) [reply cypress/integration/examples/app.spec.js:168]: Good point! Added!

**Pull request merged**